packages/scaffold-config/src/frameworks.ts

```diff
@@ -75,20 +72,22 @@ const reactMountModule = async (projectPath: string) => {
   return semver.major(reactPkg.detectedVersion) === 18 ? 'cypress/react18' : 'cypress/react'
 }
 
-export const WIZARD_FRAMEWORKS = [
+export const supportStatus = ['alpha', 'beta', 'full', 'community'] as const
+
+export const CT_FRAMEWORKS: ComponentFrameworkDefinition[] = [
   {
     type: 'reactscripts',
     configFramework: 'create-react-app',
     category: 'template',
     name: 'Create React App',
-    supportedBundlers: [dependencies.WIZARD_DEPENDENCY_WEBPACK],
+    supportedBundlers: ['webpack'],
     detectors: [dependencies.WIZARD_DEPENDENCY_REACT_SCRIPTS],
-    dependencies: (bundler: WizardBundler['type'], projectPath: string): Promise<DependencyToInstall[]> => {
-      return Promise.all([
-        isDependencyInstalled(dependencies.WIZARD_DEPENDENCY_REACT_SCRIPTS, projectPath),
-        isDependencyInstalled(dependencies.WIZARD_DEPENDENCY_REACT_DOM, projectPath),
-        isDependencyInstalled(dependencies.WIZARD_DEPENDENCY_REACT, projectPath),
-      ])
+    dependencies: (bundler: WizardBundler['type']): Cypress.CypressComponentDependency[] => {
```

Comment:
This is probably the main change - instead of having these objects declare the metadata **and** check for the dependencies, they are purely *definitions* - thus the new and more accurate name, `ComponentFrameworkDefinition`.

This is the same interface third parties will use, except we [omit a few](https://github.com/cypress-io/cypress/pull/25713/files#diff-dce5144c067a639808029e38cef9e4e90a64dec50aeec74e6064f3cb31a56981R293):

```
type ThirdPartyComponentFrameworkDefinition = Omit<ComponentFrameworkDefinition, 'glob' | 'codeGenFramework' | 'supportStatus' | 'specPattern'>
``

Here's why:

- `supportStatus` - these are forced to be "community" for third party integrations.
-  'glob' | 'codeGenFramework' | 'specPattern' - these are all used for Create From Component. We might support this for third parties one day, but not right now. We don't even support it for some first parties, like Svelte, for example.